benchmarks/Benchmarks.Trace/AspNetCoreBenchmark.cs

```diff
@@ -0,0 +1,141 @@
+#if !NETFRAMEWORK
+
+using System;
+using System.Net.Http;
+using System.Runtime.InteropServices;
+using System.Threading;
+using System.Threading.Tasks;
+using BenchmarkDotNet.Attributes;
+using Datadog.Trace;
+using Datadog.Trace.ClrProfiler.Integrations;
+using Datadog.Trace.Configuration;
+using Microsoft.AspNetCore.Builder;
+using Microsoft.AspNetCore.Hosting;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.AspNetCore.TestHost;
+using Microsoft.Extensions.DependencyInjection;
+
+namespace Benchmarks.Trace
+{
+    [MemoryDiagnoser]
+    public class AspNetCoreBenchmark
+    {
+        private static readonly HttpClient Client;
+
+        static AspNetCoreBenchmark()
+        {
+            Environment.SetEnvironmentVariable(ConfigurationKeys.TraceEnabled, "0");
```

Comment:
nit: I think you can set this in the TracerSettings below with `TraceEnabled = false` so you don't have to modify the environment variables

Reply:
The problem is subtle:

to initialize the tracer, I do: 

```csharp
Tracer.Instance = new Tracer(settings, new DummyAgentWriter(), null, null, null);
```
By calling `new Tracer`, the static constructor is invoked, which initializes a new tracer before I have time to construct mine. That tracer is fully functional, with background threads and stuff, and adds noise to the benchmark. Adding environment variables is the only way I found to limit what that tracer does in background. 
Now that I think about it, I should probably just add a global internal setting that prevents that tracer from being instantiated. I'll look into that in a separate PR.

Reply:
Ohh, this (tracer config in aspnet core) is the bug I'm working on right now, hehe.

Reply:
Could we move the creation of the first `Tracer` instance from the static constructor to the `Tracer.Instance` getter? If we can do this, it should allow users (including ourselves here) to set `Tracer.Instance` without creating a not-so-throwaway `Tracer` instance. Just something to think about, not for this PR.